docs/how-to-guides/upgrade-openebs.md

```diff
@@ -0,0 +1,301 @@
+  and restore.
+
+* Ensure the cluster and OpenEBS volumes are is in healthy state before proceeding.
+    ```bash
+    $ lokoctl health
+    Node                    Ready    Reason          Message
+
+    alpha-controller-0      True     KubeletReady    kubelet is posting ready status
+    alpha-large-worker-0    True     KubeletReady    kubelet is posting ready status
+    Name      Status    Message              Error
+
+    etcd-0    True      {"health":"true"}
+
+
+   $ kubectl get cstorpools -n openebs # Status should be Healthy.
+
+   NAME                                ALLOCATED   FREE   CAPACITY   STATUS    READONLY   TYPE      AGE
+   cstor-pool-openebs-replica-1-w3r7   6.98G       437G   444G       Healthy   false      striped   4d22h
+
+
+   $ kubectl get cstorvolume -n openebs # Status should be Healthy.
+
+   NAME                                       STATUS    AGE     CAPACITY
+   pvc-183dc3df-a7d9-4273-a8e2-7f66d2e19f4e   Healthy   3d20h   50Gi
+   pvc-5d4b4c2b-2ed5-4e75-aeb9-fbd2918ebb78   Healthy   3d20h   50Gi
+   ```
+## Steps
+
+### Step 1: Upgrade OpenEBS control plane components
+
+Lokomotive provides an easy way of upgrading the OpenEBS control plane.
+
+Execute the following command to upgrade OpenEBS control plane:
+
+```bash
+lokoctl component apply openebs-operator
+```
+
+Doing so, terminates the OpenEBS resources associated with `v1.12.0` and new
+resources are created with `v2.2.0`.
+
+Verify all the pods are in `Running` state before proceeding:
+
+```bash
+$ kubectl get pods -n openebs -n openebs.io/version=2.2.0
+```
+
+### Step 2: Upgrade OpenEBS data plane components
+
+OpenEBS control plane and data plane components work independently. Even if the
+control plane components are upgraded to `v2.2.0`, the data plane components
+continue to work with the older version `v1.12.0`.
+
+#### Upgrade cStor pools
+
+Get existing `StoragePoolClaims` by executing:
+
+```bash
+kubectl get spc
+NAME                           AGE
+cstor-pool-openebs-replica-3   2d21h26m
+```
+
+Create a Job to upgrade the existing cStor pools:
+
+```bash
+OPENEBS_NEW_VERSION=2.2.0
+OPENEBS_OLD_VERSION=1.12.0
+cat > upgrade-cstor-pools-1120-220.yaml <<EOF
+# upgrade-cstor-pools-1120-220.yaml
+
+#This is an example YAML for upgrading cstor SPC.
+#Some of the values below needs to be changed to
+#match your openebs installation. The fields are
+#indicated with VERIFY
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  #VERIFY that you have provided a unique name for this upgrade job.
+  #The name can be any valid K8s string for name. This example uses
+  #the following convention: cstor-spc-<flattened-from-to-versions>
+  generateName: cstor-spc-1120220-
+  namespace: openebs
+
+spec:
+  backoffLimit: 4
+  template:
+    spec:
+      serviceAccountName: openebs-operator
+      containers:
+      - name:  upgrade
+        args:
+        - "cstor-spc"
+
+        # --from-version is the current version of the pool
```

Comment:
we can get rid of all the comments in the files to make it concise. Also it is understandable without comments what a particular flag does because we have those env vars which are very descriptive.